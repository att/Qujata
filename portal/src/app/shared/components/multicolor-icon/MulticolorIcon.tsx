import React, { NamedExoticComponent, ReactNode } from 'react';
import cn from 'classnames';

export interface MulticolorIconProps {
  baseClass: string;
  nameClass: string;
  shapesAmount: number;
  className?: string;
}

export const MulticolorIcon: React.FC<MulticolorIconProps> = (props: MulticolorIconProps) => {
  if (props.shapesAmount < 1) {
    // eslint-disable-next-line no-null/no-null
    return null;
  }
  const items: ReactNode[] = [];
  for (let i: number = 0; i < props.shapesAmount; i += 1) {
    const iconPathClassName: string = `${props.nameClass}-path${i + 1}`;
    items.push(<i key={i} className={iconPathClassName} />);
  }
  return (
    <i className={cn(props.className, props.baseClass, props.nameClass)}>
      {items}
    </i>
  );
};

export const MulticolorIconMemoized: NamedExoticComponent<MulticolorIconProps> = React.memo<MulticolorIconProps>(MulticolorIcon);
