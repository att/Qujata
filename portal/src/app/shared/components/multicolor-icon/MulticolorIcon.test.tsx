import { render, RenderResult } from '@testing-library/react';
import { MulticolorIconMemoized, MulticolorIconProps } from './MulticolorIcon';

describe('MulticolorIcon', () => {
  test('renders multicolor icon with 4 nodes', () => {
    const props: MulticolorIconProps = {
      baseClass: 'eda',
      nameClass: 'xaaf-icon',
      shapesAmount: 4,
    };
    const { container }: RenderResult = render(<MulticolorIconMemoized {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should not render when receive wrong size', () => {
    const props: MulticolorIconProps = {
      baseClass: 'eda',
      nameClass: 'xaaf-icon',
      shapesAmount: 0,
    };
    const { container }: RenderResult = render(<MulticolorIconMemoized {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
