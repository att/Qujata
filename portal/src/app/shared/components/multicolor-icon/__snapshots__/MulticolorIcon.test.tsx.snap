// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MulticolorIcon renders multicolor icon with 4 nodes 1`] = `
<i
  class="eda xaaf-icon"
>
  <i
    class="xaaf-icon-path1"
  />
  <i
    class="xaaf-icon-path2"
  />
  <i
    class="xaaf-icon-path3"
  />
  <i
    class="xaaf-icon-path4"
  />
</i>
`;

exports[`MulticolorIcon should not render when receive wrong size 1`] = `null`;
